### PR TITLE
Read architecture natively

### PR DESCRIPTION
**Pull request by tropicar**
The main idea is to obtain the architecture with `uname` command. Pass this information to the buildAndUpload function and create the path due to this information.
Now check if the architecture is arm, in another case the response is the default value as before.

### CODE REVIEW

File: src/utils/getArchitecture.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Architecture } from "../types";
+import os from "os";
+
+/**
+ * Return what kind of architecture is using the machine where the build is done
+ */
+export async function getArchitecture(): Promise<Architecture> {
```
Comment (dapplion) [inline src/utils/getArchitecture.ts:7]: This doesn't have to be an async function

File: src/tasks/buildAndUpload.ts [hunk 2]
```diff
@@ -65,6 +66,7 @@ export function buildAndUpload({
   deleteOldPins?: boolean;
   composeFileName: string;
   dir: string;
+  hardwareArchitecture: Architecture;
```
Comment (dapplion) [inline src/tasks/buildAndUpload.ts:69]: Why pass this as an argument when you could compute it directly in the body of the function? Adds unnecessary code duplication

File: src/utils/getArchitecture.ts (hunk 1 above)
Comment (tropicar) [reply src/utils/getArchitecture.ts]: Fixed

File: src/tasks/buildAndUpload.ts (hunk 2 above)
Comment (tropicar) [reply src/tasks/buildAndUpload.ts]: How it was an async function, I had to obtain that value outside of the build function so this function was sync, but after changing the getArchitecture() to sync function, it's no a problem to include into the body of the build function.
Fixed

**Comment by dapplion** [comment]
Please see all the warnings about un-used variables and fixed them.

**Review by dapplion — APPROVED** [review]

**Pull request merged**